src/app/util/ember-compatibility-functions.cpp

```diff
@@ -199,6 +199,23 @@ CHIP_ERROR ReadSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVWriter * ap
                   ChipLogValueMEI(aClusterInfo.mClusterId), ChipLogValueX64(aClusterInfo.mNodeId), aClusterInfo.mEndpointId,
                   aClusterInfo.mFieldId, aClusterInfo.mListIndex);
 
+    AttributeAccessInterceptor * interceptor = findAttributeAccessInterceptor(aClusterInfo.mEndpointId, aClusterInfo.mClusterId);
+    if (interceptor != nullptr)
+    {
+        bool dataRead;
+        // TODO: We should probably clone the writer and convert failures here
+        // into status responses, unless our caller already does that.
+        ReturnErrorOnFailure(interceptor->ReadAttributeData(aClusterInfo, apWriter, &dataRead));
```

Comment:
Shouldn't we be checking if this returns some kind of 'unsupported' error to then fallback to the standard call of `emberAfReadAttribute`?

Reply:
The interceptor not wanting to handle the read is signaled by a success return and false `dataRead`.